File: nga_phone_base_3.0/src/sp/phone/lab/mvp/BaseFragment.java
package sp.phone.lab.mvp;

import android.content.Context;
import android.support.v4.app.FragmentManager;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.LayoutInflater;
import android.view.MenuItem;
import android.widget.Toast;

import com.trello.rxlifecycle2.components.support.RxFragment;

import gov.anzong.androidnga.activity.BaseActivity;

/**
 * Created by liuboyu on 16/6/28.
 */
public abstract class BaseFragment extends RxFragment {

    protected AppCompatActivity mActivity;

    protected Toast mToast;

    public void showToast(int res) {
        String str = getString(res);
        showToast(str);
    }

    public void showToast(String res) {
        if (mActivity == null){
            return;
        }
        if (mToast != null) {
            mToast.setText(res);
            mToast.setDuration(Toast.LENGTH_SHORT);
        } else {
            mToast = Toast.makeText(mActivity, res, Toast.LENGTH_SHORT);
        }
        mToast.show();
    }

    protected BaseActivity getBaseActivity() {
        return (BaseActivity) getActivity();
    }

    protected void changeNightMode(final MenuItem menu) {
        getBaseActivity().changeNightMode(menu);
    }


    public void finish(){
        if (mActivity != null) {
            mActivity.finish();
        }
    }

    protected void setSupportActionBar(Toolbar toolbar) {
        if (mActivity != null) {
            mActivity.setSupportActionBar(toolbar);
        }
    }

    protected ActionBar getSupportActionBar() {
        return mActivity != null ? mActivity.getSupportActionBar() : null;
    }

    protected void setTitle(String title) {
        if (mActivity != null) {
            mActivity.setTitle(title);
        }
    }

    protected void setTitle(int resId) {
        setTitle(getString(resId));
    }

    protected LayoutInflater getLayoutInflater() {
        return mActivity != null ? mActivity.getLayoutInflater() : null;
    }

    protected FragmentManager getSupportFragmentManager() {
        return getChildFragmentManager();
    }

    @Override
    public void onAttach(Context context) {
        mActivity = (AppCompatActivity) context;
        super.onAttach(context);
    }

    @Override
    public void onDetach() {
        mActivity = null;
        super.onDetach();
    }
}

File: nga_phone_base_3.0/src/main/java/gov/anzong/androidnga/ui/fragment/BaseFragment.kt
package gov.anzong.androidnga.ui.fragment

import androidx.annotation.LayoutRes
import androidx.appcompat.widget.Toolbar
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.lifecycle.ViewModelProvider
import gov.anzong.androidnga.R
import gov.anzong.androidnga.activity.BaseActivity

open class BaseFragment : Fragment {

    private val mActivityViewModelProvider: ViewModelProvider by lazy { ViewModelProvider(activity as FragmentActivity) }

    constructor(@LayoutRes contentLayoutId: Int) : super(contentLayoutId)

    constructor() : super()

    protected fun initToolbar() {
        val toolbar: Toolbar? = view?.findViewById(R.id.toolbar);
        if (toolbar != null && activity != null) {
            (activity as BaseActivity).setupToolbar(toolbar)
        }
    }

    protected fun setTitle(title: String) {
        activity?.title = title
    }

    protected fun getActivityViewModelProvider(): ViewModelProvider {
        return mActivityViewModelProvider;
    }

    open fun onBackPressed() : Boolean {
        return false
    }
}
